akka-javasdk/src/main/java/akka/javasdk/consumer/MessageContext.java

```diff
@@ -22,4 +21,21 @@ public interface MessageContext extends MetadataContext {
 
   /** Access to tracing for custom app specific tracing. */
   Tracing tracing();
+  /**
+   * Returns true if this message was originated on the same region that is being processed.
+   *
+   * <p>For messages coming from Akka entities:
+   *
+   * <ul>
+   *   <li>If the message is an event from an Event Sourced Entity, it returns true if the event was
+   *       originated in the region where this consumer is running. Otherwise, it returns false.
+   *   <li>If the message is a change update from a KeyValueEntity, it returns true if the update
+   *       was originated in the region where this consumer is running. Otherwise, it returns false.
+   *   <li>For messages coming from any broker (e.g., Kafka or Google PubSub), this method always
+   *       returns false.
```

Comment:
could be worth mentioning that it's also always false for events from another service (`@Consume.FromServiceStream`)

Reply:
Will that be the case?
In the runtime, we do have origin information and it's being added to the message as well. 

Let's say that ServiceA running in region 1 and 2 is exposing a stream. The stream in both regions will contain the `originRegion` info. 

ServiceB is consuming the stream from ServiceA. The consumer in region1 and in region2 will both consume the stream from their respective regions, ie: B1 consumes A1 and B2 consumes A2, but both streams will contain messages originated in the other region.

Reply:
ReplicatedEventMetadata is not propagated in Projections over gRPC, which is what we use for service-to-service eventing.

ReplicatedEventMetadata is used for Replicated Event Sourcing.